great_expectations/datasource/fluent/file_path_data_asset.py

```diff
@@ -140,6 +164,19 @@ def __init__(self, **data):
         )
         self._all_group_names = self._regex_parser.get_all_group_names()
 
+    def get_partitioner_implementation(
+        self, abstract_partitioner: Partitioner
+    ) -> SparkPartitioner:
+        PartitionerClass = self._partitioner_implementation_map.get(
+            type(abstract_partitioner)
+        )
+        if PartitionerClass is None:
+            raise ValueError(
+                f"Requested Partitioner `{abstract_partitioner.method_name}` is not implemented for this DataAsset. "
+            )
+        assert PartitionerClass is not None
```

Comment:
Do we need this assert?

Reply:
i originally added this because mypy wasn't correctly catching that the exception above narrows the type to non null, but i'll double check 🙇